python/ray/autoscaler/node_provider.py

```diff
@@ -134,7 +134,7 @@ def _load_class(path):
 
 
 def _get_node_provider(provider_config: Dict[str, Any],
-                       cluster_name: str) -> Any:
+                       cluster_name: str) -> Union[Any, 'NotImplemented']:
```

Comment:
```suggestion
                       cluster_name: str) -> Union[Any, "NotImplemented"]:
```

Reply:
How would you recommend handling the methods which are not implemented yet, e.g. NodeProvider.is_running()?
https://github.com/ray-project/ray/pull/11023/files#diff-c732efe3edd32eb2ec55d17741db1c12L189